docs/api/standard_library_types.md

```diff
@@ -152,6 +152,117 @@ print(m.model_dump())
 #> {'td': datetime.timedelta(days=3, seconds=45005)}
 ```
 
+## List
+
+### `list`
+
+Allows `list`, `tuple`, `set`, `frozenset`, `deque`, or generators and casts to a list.
+when a generic parameter is provided, the appropriate validation is applied to all items of the list.
+
+### `typing.List`
+Handled the same as `list` above
+
+```py
+from typing import List, Optional
+
+from pydantic import BaseModel
+
+
+class Model(BaseModel):
+    simple_list: Optional[list] = None
+    list_of_ints: Optional[List[int]] = None
+
+
+print(Model(simple_list=['1', '2', '3']).simple_list)
+#> ['1', '2', '3']
+print(Model(list_of_ints=['1', '2', '3']).list_of_ints)
+#> [1, 2, 3]
+```
+
+## Tuple
+
+### `tuple`
+Allows `list`, `tuple`, `set`, `frozenset`, `deque`, or generators and casts to a tuple.
+when generic parameters are provided, the appropriate validation is applied to the respective items of the tuple
+
+### `typing.Tuple`
+Handled the same as `tuple` above
+
+```py
+from typing import Optional, Tuple
+
+from pydantic import BaseModel
+
+
+class Model(BaseModel):
+    simple_tuple: Optional[tuple] = None
+    tuple_of_different_types: Optional[Tuple[int, float, bool]] = None
+
+
+print(Model(simple_tuple=[1, 2, 3, 4]).simple_tuple)
+#> (1, 2, 3, 4)
+print(Model(tuple_of_different_types=[3, 2, 1]).tuple_of_different_types)
+#> (3, 2.0, True)
+```
+
+## Deque
+
+### `deque`
+Allows `list`, `tuple`, `set`, `frozenset`, `deque`, or generators and casts to a `deque`.
+when generic parameters are provided, the appropriate validation is applied to the respective items of the `deque`
+
+### `typing.Deque`
+Handled the same as `deque` above
+
+```py
+from typing import Deque, Optional
+
+from pydantic import BaseModel
+
+
+class Model(BaseModel):
+    deque: Optional[Deque[int]] = None
+
+
+print(Model(deque=[1, 2, 3]).deque)
+#> deque([1, 2, 3])
+```
+
+## NamedTuple
+
+### `subclasses of typing.NamedTuple`
+Similar to `tuple`, but creates instances of the given `namedtuple` class.
+
+### `types returned from collections.namedtuple`
+Similar to `subclass of typing.NamedTuple`, but since field types are not specified,
+all fields are treated as having type `Any`.
```

Comment:
```suggestion
### Types returned from `collections.namedtuple`

Similar to `subclass of typing.NamedTuple`, but since field types are not specified,
all fields are treated as having type `Any`.
```